lib/utils.js

```diff
@@ -67,17 +67,59 @@ function addArrayToSet(array, set, valueExtractor) {
   return set
 }
 
+function extractLicenseFromLicenseUrl(licenseUrl) {
+  if (!licenseUrl) return
+  for (const licenseUrlOverride of _licenseUrlOverrides) {
+    const licenseUrlMatch = licenseUrlOverride.test.exec(licenseUrl)
+    if (licenseUrlMatch) {
+      if (licenseUrlOverride.license) {
+        return licenseUrlOverride.license
+      } else if (licenseUrlOverride.licenseMatchGroup) {
+        return licenseUrlMatch[licenseUrlOverride.licenseMatchGroup]
+      }
+    }
+  }
+}
+
 function _normalizeVersion(version) {
   if (version == '1') return '1.0.0' // version '1' is not semver valid see https://github.com/clearlydefined/crawler/issues/124
   return semver.valid(version) ? version : null
 }
 
+const _licenseUrlOverrides = [
+  {
+    test: /^\w*https?:\/\/(?:www.)?opensource.org\/licenses\/mit-license(\.php)?$/i,
+    license: 'MIT'
+  },
+  {
+    test: /^\w*https?:\/\/(?:www.)?opensource.org\/licenses\/(.*?)(\.php)?$/i,
+    licenseMatchGroup: 1
+  },
+  {
+    test: /^\w*http(s)?:\/\/(www.)?apache.org\/licenses\/LICENSE-2\.0(.html)?$/i,
+    license: 'Apache-2.0'
+  },
+  {
+    test: /^\w*https?:\/\/(?:www.)?gnu.org\/licenses\/(.*?)(\.html)?$/i,
+    licenseMatchGroup: 1
+  },
+  {
```

Comment:
This one (and the one below) are Microsoft specific so we shouldn't include them in ClearlyDefined.